website/pages/blog/introducing-the-gandi-source-plugin.md

```diff
@@ -0,0 +1,99 @@
+---
+title: Introducing The Gandi Source Plugin
+tag: announcement
+date: 2022/12/01
+description: >-
+CloudQuery is an open-source, extensible framework that gives you a
+single-pane-of-glass to your cloud-infrastructure using SQL. Today, we are
+happy to announce the release of the Gandi source plugin CloudQuery.
+author: disq
+---
+
+import { BlogHeader } from "../../components/BlogHeader"
+
+<BlogHeader/>
+
+CloudQuery is an open-source, extendable framework that gives you a single-pane-of-glass to your cloud-infrastructure using SQL. Today, we are happy to announce the release of the Gandi source plugin for CloudQuery.
+
+Gandi is a company providing domain name registration, web hosting and related services. Gandi also provides a cloud platform, which includes a virtual private server (VPS) offering.
+
+CloudQuery now supports pulling domain and DNS resources from [Gandi](https://gandi.net/) using Gandi's API.
+
+Take a look at our [Configuration](/docs/plugins/sources/gandi/configuration) section to configure required credentials for the plugin.
+
+Below are some real world query examples to get you started.
+
+## Detect soon to expire domain registrations
+
+If you have more than just one domain, keeping track of which one will expire soon can be a problem. Here's a query to check how many days left on each registration.
+
+```sql
+select fqdn, dates->>'registry_ends_at' as registry_ends_at, date_trunc('day', (dates->>'registry_ends_at')::timestamp - current_timestamp) as days_left from gandi_domains where ((dates->>'registry_ends_at')::timestamp - interval '90 day') < current_timestamp order by 1;
+```
+
+This query would output a table of domain names that are going to expire in the next 90 days or less:
+
+```
+      fqdn      |   registry_ends_at   | days_left
+----------------+----------------------+-----------
+ yourdomain.co  | 2023-01-28T08:09:41Z | 58 days
+ yourdomain.com | 2023-01-31T15:58:08Z | 61 days
+(2 rows)
+```
+
+You can now put this in a dashboard or set up an alert on it.
+
+## Make sure all your LiveDNS domains have automatic snapshots
+
+Automatic snapshots is a Gandi LiveDNS feature to ensure you always have a back up to a previous state of the domain configuration. With the following query it's possible to list any domains you manage but which don't have automatic snapshots enabled for some reason.
+
+```sql
+select d.fqdn from gandi_livedns_domains d left join gandi_livedns_snapshots s on s.fqdn=d.fqdn and s.automatic where s.fqdn is null;
+```
+
+This will return a list of domain names that don't have the automatic snapshots feature enabled. You can set up on alert on this query and make sure your domain configurations are automatically backed up.
```

Comment:
```suggestion
This will return a list of domain names that don't have the automatic snapshots feature enabled. You can set up an alert on this query and make sure your domain configurations are automatically backed up.
```